src/utilities/transformSchema.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type { ObjMap } from '../jsutils/ObjMap';
+import keyValMap from '../jsutils/keyValMap';
+import invariant from '../jsutils/invariant';
+import type { GraphQLSchemaConfig } from '../type/schema';
+import { GraphQLSchema } from '../type/schema';
+import { isSpecifiedScalarType } from '../type/scalars';
+import { isIntrospectionType } from '../type/introspection';
+import type { GraphQLDirectiveConfig } from '../type/directives';
+import { GraphQLDirective } from '../type/directives';
+import { GraphQLList, GraphQLNonNull } from '../type/wrappers';
+import type {
+  GraphQLNamedType,
+  GraphQLScalarTypeConfig,
+  GraphQLObjectTypeConfig,
+  GraphQLInterfaceTypeConfig,
+  GraphQLUnionTypeConfig,
+  GraphQLEnumTypeConfig,
+  GraphQLInputObjectTypeConfig,
+} from '../type/definition';
+import {
+  GraphQLScalarType,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLUnionType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  isListType,
+  isNonNullType,
+  isScalarType,
+  isObjectType,
+  isInterfaceType,
+  isUnionType,
+  isEnumType,
+  isInputObjectType,
+} from '../type/definition';
+
+type ExactSchemaConfig = $Exact<GraphQLSchemaConfig> & {
+  types: Array<*>,
```

Comment:
We should avoid `*` as much as possible - Flow often cannot resolve them, so these are either effectively all `any` or they may be derived to some type which could produce an error that is very difficult to understand. They appear a lot in this file, I think almost every case should be able to be replaced with a true type reference